### DISCUSSION TITLE
Potentially switch from chokidar to @parcel/watcher or turbowatch or `fs.watch`

### DISCUSSION THREAD

**Discussion by uncenter** [11ty/eleventy#4025] [Enhancement Queue]
https://github.com/parcel-bundler/watcher seems like a much better alternative to Choikdar. It's already used by projects like Tailwind, Nx, Nuxt, VSCode, etc (https://github.com/parcel-bundler/watcher#who-is-using-this). Vite [considered switching to it](https://github.com/vitejs/vite/issues/13593) but it happened to be too large of a milestone before Vite 5. On that note, https://github.com/gajus/turbowatch also seems interesting. 👀

**Comment by zachleat** [comment]
I’m assuming this should be moved to `11ty/eleventy`—the chokidar usage here is strictly for the standalone dev-server CLI. If we change it upstream in Eleventy core, we can talk about applying it to the dev server CLI too!

**Comment by uncenter** [comment]
Ah okay, makes sense! I thought that was the main watcher in general. Thanks for the correction.

**Comment by zachleat** [comment]
Whoa, just saw https://github.com/parcel-bundler/watcher?tab=readme-ov-file#querying which would be a pretty big win for incremental builds.

**Comment by murtuzaalisurti** [comment]
does @parcel/watcher support polling though? https://github.com/parcel-bundler/parcel/issues/564

**Comment by zachleat** [comment]
Good callout. I have access to WSL for testing, if/when I tackle this one I’ll make sure to test appropriately.

**Comment by zachleat** [comment]
Also a shout out to `fs.watch` #3125. Going to keep this one open as home-base
